lib/utils.rb

```diff
@@ -31,5 +32,31 @@ def Utils.paramhash_to_str params
   params.collect {|k, v| if (v == true) then ";#{k}" else ";#{k}=#{v}" end}.join("")
 end
 
+def Utils.check_route_matches_proxy route, proxy
+  regexp = /<sip:(.*?)@(.*?):(.*?);(.*?)>/i
+  proxy_address = proxy.instance_variable_get(:@ip)
+  proxy_port = proxy.instance_variable_get(:@port)
+  if route =~ regexp
+    route_address = $2
+    route_port = $3
+  end
+
+  route_ip = []
+  Resolv.each_address(route_address) do |address|
+    route_ip << address
+  end
+
+  proxy_ip = []
+  Resolv.each_address(proxy_address) do |address|
+    proxy_ip << address
+  end
+
+  if (route_ip & proxy_ip).empty?
```

Comment:
Worth a comment to clarify what you're doing here - I see what it is (set union to check that at least one IP address is in both sets) but I had to think about it.